addons/phone_validation/models/mail_thread_phone.py

```diff
@@ -65,13 +66,14 @@ def init(self):
                          tablename=self._table,
                          expressions=[regex_expression],
                          where=f'{fname} IS NOT NULL')
-            # The trigram index covers operators 'like', 'ilike' and '=like' starting with a wildcard
-            create_index(self.env.cr,
-                         indexname=f'{self._table}_{fname}_partial_gin_idx',
-                         tablename=self._table,
-                         method='gin',
-                         expressions=[regex_expression + ' gin_trgm_ops'],
-                         where=f'{fname} IS NOT NULL')
+            if has_trigram(self.env.cr):
```

Comment:
This won't change during the loop. This can be called only once before the loop.

Reply:
True, will adjust that.

Reply:
Coudn't we simply use `self.env.registry.has_trigram` ?

Reply:
Ok just checked, the field `has_trigram` is set during the `init` of the registry, which is done in it's `new` which is called in `preload_registries`, called just after server launching, way before any module is trying to be loaded. So it's safe to rely on `has_trigram` (the attribute) on the registry to avoid the query done by `has_trigram()`.
Will update the patch. Tnx for letting me know @xavieralt.